packages/sdk/src/convertCredentials.ts

```diff
@@ -198,3 +205,19 @@ export function protocolToConnectorType(protocol: string): ConnectorType {
 
   throw new Error(`Unknown protocol ${protocol}`)
 }
+
+/**
+ * Convert a protocol to the equivalent database connector type.
+ * Returns `undefined` if the protocol is not recognized.
+ * @param protocol e.g., 'postgres:'
+ */
+export function safeProtocolToConnectorType(protocol: string): ConnectorType | undefined {
+  const provider = pipe(
+    E.tryCatch(
+      () => protocolToConnectorType(protocol),
+      (_) => undefined,
+    ),
+    E.toUnion,
+  )
+  return provider
+}
```

Comment:
This new function and additional tests are not needed as it will never be executed in `DbPush.ts` if the protocol is not known, because code will return with exit 1 before that, which mean it will never throw here.
We could improve things I guess, but this seems not needed at the moment.